### ISSUE TITLE
`array-bracket-newline`: Allow support for both `consistent` and `{"multiline": true}` together

### ISSUE THREAD

**Issue by shixianqin** [eslint-stylistic/eslint-stylistic#799]
### What rule do you want to improve?

array-bracket-newline

### What it does?

Currently, the rule supports either:

- `"consistent"`: enforce either always in a single line or always in multiple lines if at least one bracket is broken.
- `{ "multiline": true }`: enforce line breaks if the array spans multiple lines.

However, I want both behaviors to work at the same time.

### What problem are you trying to solve?

I want to enforce that:

1. If I manually break an array element into a new line, the brackets must also be in new lines (like multiline).
2. And when array brackets are broken, their padding is consistent — i.e., if the opening bracket is on a new line, the closing must be too (like consistent).
3. But I don't necessarily want to force line breaks if the array is short and inline.

Unfortunately, currently, we can only configure either "consistent" or { "multiline": true }, but not both at the same time.

### Code Examples

✅ Expected behavior:

These should be valid:

```js
const list = ['a'];
const list = [
  'a',
];
const list = [
  'a',
  'b',
];
```

❌ These should be reported (inconsistent bracket line breaks):

```js
const list = [
  'a'];
const list = ['a',
];
```

Right now, either config alone cannot enforce this behavior.

### Proposed Solution

Could we allow a configuration like:

```js
"array-bracket-newline": ["error", {
  "multiline": true,
  "consistent": true
}]
```

or maybe a new preset option like:

```js
"array-bracket-newline": ["error", "multiline-consistent"]
```

That would enforce:

- Only require bracket line breaks when array values span multiple lines
- But when line breaks are present, the bracket position must be consistent

Thanks for your great work maintaining this project! This feature would help keep multiline formatting predictable and more stylistic.

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by ShGKme** [comment]
- Related (duplication?): https://github.com/eslint-stylistic/eslint-stylistic/issues/761

**Comment by shixianqin** [comment]
Thanks for reminding me! It was my oversight! I will close this issue
